### PR TITLE
update `versions` dependency to version 6

### PR DESCRIPTION
**Pull request by peterbecich**
### Description of the change

Trying to fix Spago in Nixpkgs: https://github.com/NixOS/nixpkgs/issues/253198

Nixpkgs provides only `versions` version 6: https://search.nixos.org/packages?channel=unstable&show=haskellPackages.versions&from=0&size=50&sort=relevance&type=packages&query=versions

`versions` changelog: https://hackage.haskell.org/package/versions-6.0.1/changelog

### Checklist:

- [x] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [ ] Added a test for the contribution (if applicable)

**P.S.**: the above checks are not compulsory to get a change merged, so you may skip them. However, taking care of them will result in less work for the maintainers and will be much appreciated 😊

### CODE REVIEW

File: test/BumpVersionSpec.hs [hunk 1]
```diff
@@ -107,18 +107,18 @@ spec = describe "spago bump-version" $ do
...
 
       -- TODO is this desired behavior, or should we just drop ONE 'v'? I'd agree it's edge case, but still :-)
       it "should drop multiple 'v's from the beginning" $
-        parseVersionBump "vvvvvvvv1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] Nothing))
+        parseVersionBump "vvvvvvvv1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 Nothing Nothing))
 
       -- TODO should this work or should we strip these in parser implementation?
       it "should parse versions with PREREL and META tags" $
-        parseVersionBump "1.2.3-r1+git123" `shouldBe` Just (Exact (SemVer 1 2 3 [[Str "r", Digits 1]] (Just "git123")))
+        parseVersionBump "1.2.3-r1+git123" `shouldBe` Just (Exact (SemVer 1 2 3 (Just (Release (pure (Alphanum "r1")))) (Just "git123")))
```
Comment (peterbecich) [inline test/BumpVersionSpec.hs:121]: Unsure if this should be `Alphanum "r1"` or `Numeric 1`

https://hackage.haskell.org/package/versions-6.0.1/docs/Data-Versions.html#t:Chunk

File: src/Spago/Version.hs [hunk 2]
```diff
@@ -60,8 +60,9 @@ getCurrentVersion = do
 
   case Safe.maximumMay tags of
     Nothing -> do
-      logInfo $ display $ "No git version tags found, so assuming current version is " <> unparseVersion mempty
-      pure mempty
+      let currentVersion = SemVer 0 0 0 Nothing Nothing
```
Comment (peterbecich) [inline src/Spago/Version.hs:63]: `SemVer` constructor has changed. Guessing this is the correct value.

**Review by f-f — APPROVED** [review]
